packages/asset/contracts/mock/TestERC20.sol

```diff
@@ -1,3 +1,30 @@
 //SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
-import {TestERC20} from "@sandbox-smart-contracts/dependency-royalty-management/contracts/mock/TestERC20.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "hardhat/console.sol";
+
+contract TestERC20 is ERC20 {
+    constructor(string memory name_, string memory symbol_) ERC20(name_, symbol_) {}
+
+    function mint(address account, uint256 amount) external {
+        _mint(account, amount);
+    }
+
+    /// @notice Approve `target` to spend `amount` and call it with data.
+    /// @param target The address to be given rights to transfer and destination of the call.
+    /// @param amount The number of tokens allowed.
+    /// @param data The bytes for the call.
+    /// @return The data of the call.
+    function approveAndCall(
+        address target,
+        uint256 amount,
+        bytes calldata data
+    ) external payable returns (bytes memory) {
+        _approve(_msgSender(), target, amount);
+
+        // solhint-disable-next-line avoid-low-level-calls
+        (bool success, bytes memory returnData) = target.call{value: msg.value}(data);
+        require(success, string(returnData));
+        return returnData;
+    }
+}
```

Comment:
this is actually an EIP https://eips.ethereum.org/EIPS/eip-1363

maybe you can find something on OZ
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/interfaces/IERC1363.sol
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/mocks/token/ERC1363ForceApproveMock.sol

Reply:
Are you saying that instead of having the function added manually we should inherit the whole ERC1363 on the TestERC20? Why?

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/ERC1363.sol

Reply:
i meant using the one from OZ instead of creating ur own mock